lib/CMakeLists.txt

```diff
@@ -26,11 +26,7 @@ if(BUILD_UNBUNDLED)
     find_package(up-cpp REQUIRED)
     find_package(zenohc REQUIRED)
 else()
-    if(CROSS_COMPILE)
-        find_package(zenohc REQUIRED)
-    else()
-        find_library(ZENOH_LIB zenohc)
-    endif()
+    find_library(ZENOH_LIB zenohc)
     find_package(up-cpp QUIET)
```

Comment:
This doesn't appear to work with the new changes to conanfile.py. It is now necessary to set `build_unbundled=False` due to the missing zenoh package. This means that up-cpp _will not be a package_ that can be found in this way.

Before the changes to conanfile.py, it was possible to use `build_unbundled=True` and `zenoh_package=False` for `conan install`, then set `BUILD_UNBUNDLED=True` for cmake. That kept up-cpp as a conan dependency, but pulled zenoh-c from the system.

Reply:
@maxfaine  as greg mentioned , this changes were made to support full build